### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by morvy** [XjSv/Cooked#28]
Since installing Cooked, the page load times have doubled on my site. It's fine when the content is cached, but admin is not cached and is much slower. As soon as I disable Cooked, admin is as fast as before. I also checked the performance with Code Profiler Pro and it shows Cooked to take more time to load classes than WooCommerce, Yoast and WP Rocket combined.

`/wp-content/plugins/cooked/vendor/composer/ClassLoader.php` takes 0.78s to load, while Woo takes only 0.15s.

**Comment by morvy** [comment]
I've added the optimize-autoloader, dump-autoload -o and uploaded. The site crashed, because math-executor is not very nice about versioning :) 2.3.2 is compatible with PHP7.4, 2.3.3 with 8.0. Such breaking change shouldn't be a patch version. So I restricted the library to 2.3.2 as I cannot upgrade PHP on this specific site. Optimized autoloader helped a bit, now it's down to 0.64s but it's still a lot.

**Comment by XjSv** [comment]
I'm wondering if this is just the `nxp/math-executor` library that is causing this. Can you possibly try to remove the dependency on that and run the profiler again?

Basically just comment out the:
```require_once __DIR__ . '/vendor/autoload.php';```
In line # 31 in `wp-content/plugins/cooked/cooked.php`

And in `wp-content/plugins/cooked/includes/class.cooked-measurements.php` comment out:

Line # 13 
```use NXP\MathExecutor;```
And Line # 496 & # 470
```
$mathExecutor = new MathExecutor();
$o = $mathExecutor->execute($expression);
```

**Comment by morvy** [comment]
Well,
without nxp, the load time went down to 0.01s.

Is it used for calculating ingredients when you change yields? If the functionality is not used, it would be great to skip whole loading of the library, at least with a wp-config constant or a hook. I understand some people still need it, but on my site it's not used and removing it helped a lot.

Thanks

**Comment by XjSv** [comment]
Yeah I suspected it to be the issue but was not sure. In that case I'm going to see what can be done about it. Like you said, either deferring the loading of the library somehow.

**Comment by XjSv** [comment]
@morvy Hey, I'm wondering if the performance has increased at all with the latest version [v1.8.0](https://github.com/XjSv/Cooked/releases/tag/v1.8.0).

There were alot of functions converted from using `esc_html__` to `__` where the escape functionality was not needed. Was just wondering how much of an impact that had.

**Comment by morvy** [comment]
Hi,
we found out that without math-executor there were missing info, so I rewrote all measurements with strings and the page is fast, no issues. I had to replace rating module with a different solution, because customer wanted to have anonymous ratings. I was able to inject kk-stars into your plugin, so I'm happy everything works. Current version is definitely not slowing down the site more than other plugins.
